### ISSUE TITLE
GTFS RT Validations: service alerts files not being found by external table URI definition

### ISSUE THREAD

**Issue by lauriemerrell** [cal-itp/data-infra#1180]
Our `gtfs_rt.validation_trip_updates` table is essentially empty, and we suppressed the creation of `gtfs_rt.validation_service_alerts`, because our external table definition (trip updates example: https://github.com/cal-itp/data-infra/blob/main/airflow/dags/rt_loader/external_validation_trip_updates.sql#L25) is seemingly not finding any files (for service alerts, we thought no files existed). However, valid files exist for both feed types.

Specific example: `110.1`. All three validation files exist and are populated for 3/6:
![image](https://user-images.githubusercontent.com/55149902/157059993-eb2b86fb-dbe1-4b6d-85c9-689c75f2b2cc.png)

However, only the `vehicle_positions` table is successfully created and reflects this data.

**Comment by atvaccaro** [comment]
I've started digging into this today; it's tricky to really compare gcsfs globbing against bigquery URI wildcards as I don't think they are really apples-to-apples. I'm probably going to timebox this for today so I don't prevent myself from progressing on other projects.

**Comment by atvaccaro** [comment]
I am closing this as unable to reproduce. It didn't occur last night, and now service alerts are working as well. Will re-open if it occurs again.

**Comment by lauriemerrell** [comment]
Issue happened again yesterday and today; seems like it's service alerts.

We *know* that there are files present for service alerts violations (for example, see the screenshot in the original issue description). We can't figure out why the external table definition isn't finding them; it should find the files every day.

**Comment by lauriemerrell** [comment]
cc @atvaccaro 

We think that the underlying issues here will ultimately be best ameliorated by the longer-term work on bucket keying and the dbt transition. 

In the meantime, we have some ideas for things to try: 
1. Removing the dependency on the `rt_loader` DAG; this dependency is not necessary (since the view does not depend on `execution_date`) -- we will try this first to just get a sense for how often this issue is occurring, right now a lot of the failures are happening because the `wait_for` task is failing so it's hard to assess
2. Potentially suppressing the `service alerts` again (commenting it out), since that one seems the buggiest and the other two seem to be working 
   a. If we can get it to run for `trip updates` and `vehicle positions`, we can maybe see if it's actually finding all the files we expect which might help us understand what (if anything) is going on with the globbing that's failing for `service alerts`

**Comment by atvaccaro** [comment]
Ok so I figured this out, and it wasn't a globbing issue.

the tl;dr is that the RT validator pipeline writes parquet files _from empty dataframes without defined columns_ in the event that no validation errors were output. This isn't a problem for vehicle positions and trip updates, as there's quite a few errors that generally pop up at least once per day, so we always have at least a handful of rows. But service alerts is a tiny volume in comparison, and has many fewer validation rules (I think).

Those empty dataframes do have one column: the empty index i.e. `__index_level_0__`; I assume BigQuery must not be able to handle parquet files with non-overlapping columns, so it's only reading in the empties.

How I debugged and fixed this:
```python
# find all files read by the table
files = fs.glob('gs://gtfs-data/rt-processed/validation/**/service_alerts.parquet')
fs = get_fs()
results = list(tqdm((fs.get(f"gs://{file}", file) for file in files), total=len(files)))

# verified we had mismatching columns
dfs = [pd.read_parquet(file) for file in tqdm(files)]
set(tuple(df.columns.values.tolist()) for df in dfs)

# then found the empties and moved them
empty_files = [file for file in tqdm(files) if pd.read_parquet(file).empty]
for file in tqdm(empty_files):
    fs.mv(file, file.replace('service_alerts', 'service_alerts_backup.parquet')) # I accidentally added the extension here, but leaving for posterity
```

I've verified that we can select from the table now: `select calitp_itp_id from `cal-itp-data-infra-staging`.gtfs_rt.validation_service_alerts limit 100;`

I will be putting up a PR soon so the validator does not write empty dataframes to GCS, which will fix the actual cause.

**Comment by holly-g** [comment]
Fixed by https://github.com/cal-itp/gtfs-rt-validator-api/pull/7 🎉  
Unblocks #570
